doc/design/design_support_arbitrary_select.md

```diff
@@ -0,0 +1,57 @@
+# Support Arbitrary Select Statements
+
+SQLFlow extends the SQL syntax to enable model training and inference. The syntax extension should be easy to learn and integrate well with the existing SQL syntax. While the standard SQL syntax supports complicated select statements, SQLFlow should support training/predict on these statements via appending a TRAIN/PREDICT clause right after them.
+
+With the support of arbitrary select statements, a user can integrate AI into his/her database with ease.
+
+## Overview
+
+Given a long and nested select inside an extended SQL statement like
+
+```sql
+-- select clause
+SELECT c1, c2, id, class FROM (SELECT * FROM my_table) a
+-- train clause
+TRAIN DNNClassifier
+WITH n_class = 3
+COLUMN c1, c2, EMBEDDING(id)
+LABEL class
+INTO my_dnn_model;
+```
+
+SQLFlow does the following steps.
+
+1. SQLFlow splits the extended SQL statement into its select clause and its train clause. In the above example, the select clause is `SELECT ... a` and the train clause is `TRAIN DNNClassifier ... INTO my_dnn_model`. For the train clause, we check the syntax by our parser at `pkg/sql/parser.go`.
```

Comment:
Do we need to check the arbitrary select statement using third-party parsers like calcite, or directly send the statement to the DB engine? It's easy to implement if we can directly send the SQL statement to the DB engine, and if grammar error occurs, the DB engine should return some error message before the SQL is executed.

Reply:
I prefer directly sending the SQL statement to the DB engine.

The parser can check the syntax error, while the DB engine already does that.

Reply:
I see.